rdrf/registry/patients/models.py

```diff
@@ -1698,3 +1708,19 @@ def delete_associated_patient_if_any(sender, instance, **kwargs):
     if instance.relative_patient:
         if not hasattr(instance, "skip_archiving"):
             instance.relative_patient.delete()
+
+
+class PatientGUIDManager(models.Manager):
+
+    def randomString(self, letters, length):
+        return ''.join(random.choice(letters) for i in range(length))
+
+    def create(self, *args, **kwargs):
+        kwargs['guid'] = self.randomString('ABCDEFGHJKLMNPRSTUVXYZ', 6) + self.randomString('123456789', 4)
```

Comment:
Can we check this doesn't exist in the DB?
This won't fail often, but if it does it would be very inconvenient. Ex. a user going through all the registration just to fail and lose all the data they entered because a problem that isn't their fault.